File: 2571-find-the-pivot-integer/find-the-pivot-integer.py
class Solution:
    def pivotInteger(self, n: int) -> int:
        # a=0
        # b=0
        # i=1
        # x=True
        # while(x):
        #     a+=n
        #     n-=1
        #     b+=i
        #     i+=1
        #     print(a,b)
        #     if(a==b):
        #         x=False
        # print(i)

        for i in range(1,n+1):
            a=int((i*(i+1))/2)
            b=int(((n-i+1)/2)*(n+i))
            print(a,b)
            if a==b:
                return i
        return -1
            


        

File: 2571-find-the-pivot-integer/find-the-pivot-integer.cpp
class Solution {
public:
    int pivotInteger(int n) {
        int s1=0,s2=0,pivot=0;
        for(int i=1;i<=n;i++)
        {
            s1+=i;
            for(int j=i;j<=n;j++){
                s2+=j;
            }
            if(s1==s2){
            pivot=i;
            break;
            }
            s2=0;
        }
        if(pivot==0)
        return -1;
        else
        return pivot;
    }
};